File: 프로그래머스/level1/크레인 인형뽑기 게임.js
const pick = (board, col) => {
    for(let i=0; i<board.length; i++){
        if(board[i][col]!==0){
            let now = board[i][col];
            board[i][col] = 0;
            return now;
        }
    }
    return -1;
}

function solution(board, moves) {
    var answer = 0;
    let st = [];
    moves.forEach(move => {
        let now = pick(board, move-1);
        if(now!==-1){
            if(st[st.length-1]===now){
                answer += 2;
                st.pop();
            }
            else{
                st.push(now);
            }
        }
    });
    return answer;
}

File: 프로그래머스/level1/크레인 인형뽑기 게임.cpp
#include <string>
#include <iostream>
#include <vector>
#include <stack>

using namespace std;

int R;

int pick(vector<vector<int>> &board, int c){
    for(int r=0; r<R; r++){
        if(board[r][c]!=0){
            int ans = board[r][c];
            board[r][c] = 0;
            return ans;
        }
    }
    return -1;
}

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    R = board.size();
    stack<int> st;
    for(int i=0; i<moves.size(); i++){
        int result = pick(board, moves[i]-1);
        if(result==-1)  continue;
        if(!st.empty()&&st.top()==result){
            st.pop();
            answer += 2;
        }
        else{
            st.push(result);
        }
    }
    return answer;
}